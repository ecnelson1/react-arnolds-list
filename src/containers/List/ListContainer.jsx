import React, { useState, useEffect } from 'react'
import CharacterList from '../../components/Character/CharacterList';
import { allCharacters } from '../../Services/api-utils';


const ListContainer = () =>  {

    const [loading, setLoading] = useState(true);
    const [characters, setCharacters] = useState([]);

    useEffect(() => {
        allCharacters()
        .then(setCharacters)
        .finally(()=> setLoading(false));
        
    },[])

    if (loading) return <h1>LOADING...</h1>;

        return (
            <div>
                <CharacterList characters={characters} />
            </div>
        );
        };
        export default ListContainer;
