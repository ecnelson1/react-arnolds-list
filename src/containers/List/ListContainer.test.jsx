import React from 'react';
import {render, screen} from '@testing-library/react';
import {rest} from 'msw';
import {setupServer} from 'msw/node';
import ListContainer from './ListContainer';
import mockData from  '../../../fixtures/mock-data.json';

const server = setupServer(
    rest.get('https://hey-arnold-api.herokuapp.com/api/v1/characters', (req, res, ctx) => {
        return res(ctx.json(mockData));
    })
);

describe('ListContainer', () => {
    beforeAll(() => server.listen());
    afterAll(() => server.close());
    
    it('renders a loading element', async () => {
        render(<ListContainer/>);
        screen.getByText('LOADING...');
    });

    it('renders a list of characters from Hey Arnold', async ()=> {
        render(<ListContainer/>);
        const ul = await screen.findByRole('list', { name: 'characters'});
        expect(ul).toMatchSnapshot();
    });
});
