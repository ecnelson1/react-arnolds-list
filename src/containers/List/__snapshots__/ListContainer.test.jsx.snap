// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContainer renders a list of characters from Hey Arnold 1`] = `
<ul
  aria-label="characters"
>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Arnold Shortman"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f5/Arnold.png/revision/latest?cb=20181027162333"
      />
      <figcaption>
        Arnold Shortman
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Grandpa Phil"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/79/GrandpaPhil.png/revision/latest/scale-to-width-down/300?cb=20160924030436"
      />
      <figcaption>
        Grandpa Phil
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Grandma Gertie"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/7/7b/DB87AC2A-C2C2-4DAB-AECF-B9F7273C746B.png/revision/latest/scale-to-width-down/306?cb=20190816101257"
      />
      <figcaption>
        Grandma Gertie
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Mitzi"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/d/d9/Mitzi.png/revision/latest/scale-to-width-down/310?cb=20171212070331"
      />
      <figcaption>
        Mitzi
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Arnie"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/4/42/Arnie.jpg/revision/latest/scale-to-width-down/310?cb=20110109195825"
      />
      <figcaption>
        Arnie
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Miles Shortman"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
      />
      <figcaption>
        Miles Shortman
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Stella Shortman"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fa/Stella.jpg/revision/latest?cb=20101225143956"
      />
      <figcaption>
        Stella Shortman
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Grandpa's father"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e0/Grandpa%27s_father.png/revision/latest/scale-to-width-down/310?cb=20190228081112"
      />
      <figcaption>
        Grandpa's father
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Grandpa's grandfather"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/f7/Grandpa%27s_grandfather.png/revision/latest/scale-to-width-down/310?cb=20190228081015"
      />
      <figcaption>
        Grandpa's grandfather
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Oskar Kokoshka"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/c/c5/Oskar_Kokoshka.png/revision/latest/scale-to-width-down/310?cb=20180707032418"
      />
      <figcaption>
        Oskar Kokoshka
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Suzie Kokoshka"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/f/fe/Suzie.jpg/revision/latest/scale-to-width-down/310?cb=20110223003447"
      />
      <figcaption>
        Suzie Kokoshka
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Nancy"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/62/Nancy.jpg/revision/latest/scale-to-width-down/310?cb=20110423055923"
      />
      <figcaption>
        Nancy
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Ernie Potts"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/67/Ernie.jpeg/revision/latest/scale-to-width-down/310?cb=20181213181427"
      />
      <figcaption>
        Ernie Potts
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Mr. Hyunh"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/6/60/Mr._Hyunh.png/revision/latest/scale-to-width-down/310?cb=20180707031441"
      />
      <figcaption>
        Mr. Hyunh
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Mai Hyunh"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/83/Mai_Hyunh.jpeg/revision/latest/scale-to-width-down/310?cb=20190118192457"
      />
      <figcaption>
        Mai Hyunh
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Mr. Smith"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/80/Mr._Smith.jpg/revision/latest?cb=20110206233402"
      />
      <figcaption>
        Mr. Smith
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Lana Vail"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e8/762C69BE-23BA-48BA-9B14-6AF3246FCD3F.jpeg/revision/latest/scale-to-width-down/304?cb=20190106211607"
      />
      <figcaption>
        Lana Vail
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Mr. Purdy (character)"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/9/9c/Mr._Purdy_%28character%29.jpeg/revision/latest?cb=20190125214742"
      />
      <figcaption>
        Mr. Purdy (character)
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Helga G. Pataki"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/e/e5/Helga_Pataki.png/revision/latest/scale-to-width-down/266?cb=20190816021807"
      />
      <figcaption>
        Helga G. Pataki
      </figcaption>
    </figure>
  </li>
  <li>
    <figure
      aria-label="character"
    >
      <img
        alt="Robert \\"Big Bob\\" Pataki"
        src="https://vignette.wikia.nocookie.net/heyarnold/images/8/89/Big_Bob.jpg/revision/latest/scale-to-width-down/310?cb=20170428021615"
      />
      <figcaption>
        Robert "Big Bob" Pataki
      </figcaption>
    </figure>
  </li>
</ul>
`;
